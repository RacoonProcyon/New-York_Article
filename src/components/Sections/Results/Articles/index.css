#articles p{
    color: black;
    font-size: 12px;
    font-weight: 700;
}

#articles ul li{
    background-color: #F8F8F8;
    color: #000;
    font-weight: 500;
}

#articles ul li:hover{
    background-color: #397AF2;
    color: #fff;
}

#articles a{
    back
    color: blue;
    text-decoration: underline;
}