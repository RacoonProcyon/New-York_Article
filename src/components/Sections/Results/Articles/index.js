import React, { Component } from 'react'
import {Link } from 'react-router-dom'
import {connect} from 'react-redux'
import {search} from '../../../../redux/action'
class index extends Component {

    constructor(){
        super()
        this.state = {
            articlesList : []
        }
    }

    clickPrevious=(e)=>{
        e.preventDefault()
        if(this.props.pageNumber>0){
            this.props.search(this.props.recentQuery,this.props.pageNumber-1)
        }
    }

    clickNext=(e)=>{
        e.preventDefault()
        this.props.search(this.props.recentQuery,this.props.pageNumber+1)
    }

    componentDidMount = ()=>{
        this.setState({
            ...this.state,
            articlesList : this.props.searchData
        })
    }

    componentDidUpdate = (prevProps)=>{
        if(this.props.searchData !== prevProps.searchData){
            this.setState({
                ...this.state,
                articlesList:this.props.searchData
            })
        }
    }

    showList = ()=>{
        return (
            <>
            {this.state.articlesList.map((data,i)=>(
                <Link to={'/article/'+data.uri.slice(data.uri.lastIndexOf("/")+1, data.uri.length)} key={i}><li className="list-group-item">{data.headline.main}</li></Link>
            )
            )}
            </>
        )
    }

    render() {
        return (
            <article id="articles" className="col-12 col-md-8 mt-3">
                <p className="m-0">Results:</p>
                <ul className="list-group">
                    {this.state.articlesList.length?this.showList():null}
                   </ul>
                <a href='/' onClick={(e)=>this.clickPrevious(e)}>Prev</a> 
                <a className='float-right' href='/' onClick={(e)=>this.clickNext(e)}>Next</a> 
            </article>
        )
    }
}

const mapStateToProps = (state)=>{
   return {
       searchData: state.searchReducer.searchData,
       pageNumber: state.searchReducer.pageNumber,
       recentQuery: state.searchReducer.recentQuery
   }
}

const mapDispatchToProps= (dispatch)=>{
    return {
        search : (query,pageNumber)=>{dispatch(search(query,pageNumber))}
    }
}
export default connect(mapStateToProps,mapDispatchToProps)(index)
