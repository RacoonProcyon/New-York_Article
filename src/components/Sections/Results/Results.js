import React, { Component } from 'react'
import SearchBox from '../SearchBox.js'
import Header from '../Header.js'
import Articles from './Articles/index'
import PreviouslySearched from '../PreviouslySearched/PreviouslySearched.js'
import './Results.css'
import { search } from '../../../redux/action'
import { connect } from 'react-redux';

export class Results extends Component {
    componentDidMount = () => {
        this.props.search(this.props.match.params.query)
    }
    render() {
        return (
            <div>
                <Header />
                <div className="d-flex flex-column align-items-center mt-4">
                    <SearchBox currentUrl={this.props.match.params.query} history={this.props.history}/>
                    <Articles />
                    <PreviouslySearched/>
                </div>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        search: (query) => dispatch(search(query)),
    }
}

export default connect(null, mapDispatchToProps)(Results)
