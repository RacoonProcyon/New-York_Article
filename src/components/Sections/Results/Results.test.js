import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'
import  Results  from './Results';

const initialState = {
    searchData: [],
    pageNumber: 0,
    recentQuery: "",
    singleArticle: []
}
let middlewares = [thunk]
const mockStore = configureStore(middlewares)
let store = mockStore({ searchReducer: initialState })

it('renders correctly when there are no items', () => {
    const tree = renderer.create(
        <Provider store={store}>
            <Results match={{ params: { id: "123" } }}/>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});