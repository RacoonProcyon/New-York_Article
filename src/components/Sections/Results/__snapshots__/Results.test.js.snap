// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div>
  <header
    className="container-fluid px-2 py-3 d-flex flex-row justify-content-center align-items-center"
    id="header"
  >
    <p
      className="text-center text-white my-auto"
    >
      "The New York Times" article search application
    </p>
  </header>
  <div
    className="d-flex flex-column align-items-center mt-4"
  >
    <div
      className="col-12 col-md-8"
      id="searchBox"
    >
      <p
        className="m-0"
      >
        Type search query term in here:
      </p>
      <div
        className="input-group m-0"
      >
        <input
          autoComplete="off"
          autoFocus={true}
          className="form-control"
          id="queryString"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Enter article for search..."
          type="text"
        />
        <div
          className="input-group-append"
        >
          <button
            className="btn btn-warning"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-search"
            />
          </button>
        </div>
      </div>
    </div>
    <article
      className="col-12 col-md-8 mt-3"
      id="articles"
    >
      <p
        className="m-0"
      >
        Results:
      </p>
      <ul
        className="list-group"
      />
      <a
        href="/"
        onClick={[Function]}
      >
        Prev
      </a>
      <a
        className="float-right"
        href="/"
        onClick={[Function]}
      >
        Next
      </a>
    </article>
    <div
      className="dropdown px-3 col-12 col-md-8"
      id="dropdown"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        className="btn btn-secondary dropdown-toggle btn-block mt-3"
        data-toggle="dropdown"
        id="dropdownMenuButton"
        type="button"
      >
        previously searched
      </button>
      <div
        aria-labelledby="dropdownMenuButton"
        className="dropdown-menu"
        style={
          Object {
            "maxHeight": "500px",
            "overflow": "scroll",
          }
        }
      />
    </div>
  </div>
</div>
`;
